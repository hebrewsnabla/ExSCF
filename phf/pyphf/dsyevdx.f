!===============================================================================
! Copyright 2009-2019 Intel Corporation.
!
! This software and the related documents are Intel copyrighted  materials,  and
! your use of  them is  governed by the  express license  under which  they were
! provided to you (License).  Unless the License provides otherwise, you may not
! use, modify, copy, publish, distribute,  disclose or transmit this software or
! the related documents without Intel's prior written permission.
!
! This software and the related documents  are provided as  is,  with no express
! or implied  warranties,  other  than those  that are  expressly stated  in the
! License.
!===============================================================================

*
*  DSYEVD Example.
*  ==============
*
*  Program computes all eigenvalues and eigenvectors of a real symmetric
*  matrix A using divide and conquer algorithm, where A is:
*
*    6.39   0.13  -8.23   5.71  -3.18
*    0.13   8.37  -4.46  -6.10   7.21
*   -8.23  -4.46  -9.58  -9.25  -7.42
*    5.71  -6.10  -9.25   3.72   8.54
*   -3.18   7.21  -7.42   8.54   2.51
*
*  Description.
*  ============
*
*  The routine computes all eigenvalues and, optionally, eigenvectors of an
*  n-by-n real symmetric matrix A. The eigenvector v(j) of A satisfies
*
*  A*v(j) = lambda(j)*v(j)
*
*  where lambda(j) is its eigenvalue. The computed eigenvectors are
*  orthonormal.
*  If the eigenvectors are requested, then this routine uses a divide and
*  conquer algorithm to compute eigenvalues and eigenvectors.
*
*  Example Program Results.
*  ========================
*
* DSYEVD Example Program Results
* 
* Eigenvalues
* -17.44 -11.96   6.72  14.25  19.84
* 
* Eigenvectors (stored columnwise)
*  -0.26   0.31  -0.74   0.33   0.42
*  -0.17  -0.39  -0.38  -0.80   0.16
*  -0.89   0.04   0.09   0.03  -0.45
*  -0.29  -0.59   0.34   0.31   0.60
*  -0.19   0.63   0.44  -0.38   0.48
*  =============================================================================
*
*     .. Parameters ..
      INTEGER          N
      PARAMETER        ( N = 4 )
      INTEGER          LDA
      PARAMETER        ( LDA = N )
      INTEGER          LWMAX
      PARAMETER        ( LWMAX = 1000 )
*
*     .. Local Scalars ..
      INTEGER          INFO, LWORK, LIWORK
*
*     .. Local Arrays ..
      INTEGER          IWORK( LWMAX )
      DOUBLE PRECISION A( LDA, N ), W( N ), WORK( LWMAX )
      DATA             A/
     $ 0.642742149686186, 0.0d0, 0.0d0, 0.0d0,
     $-0.465821333149620,  0.337599633886206, 0.0d0, 0.0d0,
     $-0.066384044853257,  0.0481112126978231, 
     $0.0068563130848518, 0.0d0,
     $ 0.0907113787628938, -0.0657422193453504, 
     $-0.0093689020386119,  0.0128022633043168
c     $ 0.64274215, -0.46582133, -0.06638404,  0.09071138,
c     $-0.46582133,  0.33759963,  0.04811121, -0.06574222,
c     $-0.06638404,  0.04811121,  0.00685631, -0.0093689 ,
c     $ 0.09071138, -0.06574222, -0.0093689 ,  0.01280226
c     $  1.96, 0.00, 0.00, 0.00, 0.00,
c     $ -6.49, 3.80, 0.00, 0.00, 0.00,
c     $ -0.47,-6.39, 4.17, 0.00, 0.00,
c     $ -7.20, 1.50,-1.51, 5.70, 0.00,
c     $ -0.65,-6.34, 2.67, 1.80,-7.10
     $                  /
*
*     .. External Subroutines ..
      EXTERNAL         DSYEVD
      EXTERNAL         PRINT_MATRIX
*
*     .. Intrinsic Functions ..
      INTRINSIC        INT, MIN
*
*     .. Executable Statements ..
      WRITE(*,*)'DSYEVD Example Program Results'
*
*     Query the optimal workspace.
*
      LWORK = -1
      LIWORK = -1
      CALL DSYEVD( 'Vectors', 'Upper', N, A, LDA, W, WORK, LWORK,
     $             IWORK, LIWORK, INFO )
      LWORK = MIN( LWMAX, INT( WORK( 1 ) ) )
      LIWORK = MIN( LWMAX, IWORK( 1 ) )
*
*     Solve eigenproblem.
*
      CALL DSYEVD( 'Vectors', 'Upper', N, A, LDA, W, WORK, LWORK,
     $             IWORK, LIWORK, INFO )
*
*     Check for convergence.
*
      IF( INFO.GT.0 ) THEN
         WRITE(*,*)'The algorithm failed to compute eigenvalues.'
         STOP
      END IF
*
*     Print eigenvalues.
*
      CALL PRINT_MATRIX( 'Eigenvalues', 1, N, W, 1 )
*
*     Print eigenvectors.
*
      CALL PRINT_MATRIX( 'Eigenvectors (stored columnwise)', N, N, A, 
     $                   LDA )
      STOP
      END
*
*     End of DSYEVD Example.
*
*  =============================================================================
*
*     Auxiliary routine: printing a matrix.
*
      SUBROUTINE PRINT_MATRIX( DESC, M, N, A, LDA )
      CHARACTER*(*)    DESC
      INTEGER          M, N, LDA
      DOUBLE PRECISION A( LDA, * )
*
      INTEGER          I, J
*
      WRITE(*,*)
      WRITE(*,*) DESC
      DO I = 1, M
         WRITE(*,9998) ( A( I, J ), J = 1, N )
      END DO
*
 9998 FORMAT( 11(:,1X,E20.10) )
      RETURN
      END
