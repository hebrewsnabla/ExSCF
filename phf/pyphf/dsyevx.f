!===============================================================================
! Copyright 2009-2019 Intel Corporation.
!
! This software and the related documents are Intel copyrighted  materials,  and
! your use of  them is  governed by the  express license  under which  they were
! provided to you (License).  Unless the License provides otherwise, you may not
! use, modify, copy, publish, distribute,  disclose or transmit this software or
! the related documents without Intel's prior written permission.
!
! This software and the related documents  are provided as  is,  with no express
! or implied  warranties,  other  than those  that are  expressly stated  in the
! License.
!===============================================================================

*
*  DSYEV Example.
*  ==============
*
*  Program computes all eigenvalues and eigenvectors of a real symmetric
*  matrix A:
*
*    1.96  -6.49  -0.47  -7.20  -0.65
*   -6.49   3.80  -6.39   1.50  -6.34
*   -0.47  -6.39   4.17  -1.51   2.67
*   -7.20   1.50  -1.51   5.70   1.80
*   -0.65  -6.34   2.67   1.80  -7.10
*
*  Description.
*  ============
*
*  The routine computes all eigenvalues and, optionally, eigenvectors of an
*  n-by-n real symmetric matrix A. The eigenvector v(j) of A satisfies
*
*  A*v(j) = lambda(j)*v(j)
*
*  where lambda(j) is its eigenvalue. The computed eigenvectors are
*  orthonormal.
*
*  Example Program Results.
*  ========================
*
* DSYEV Example Program Results
* 
* Eigenvalues
* -11.07  -6.23   0.86   8.87  16.09
* 
* Eigenvectors (stored columnwise)
*  -0.30  -0.61   0.40  -0.37   0.49
*  -0.51  -0.29  -0.41  -0.36  -0.61
*  -0.08  -0.38  -0.66   0.50   0.40
*   0.00  -0.45   0.46   0.62  -0.46
*  -0.80   0.45   0.17   0.31   0.16
*  =============================================================================
*
*     .. Parameters ..
      INTEGER          N
      PARAMETER        ( N = 4 )
      INTEGER          LDA
      PARAMETER        ( LDA = N )
      INTEGER          LWMAX
      PARAMETER        ( LWMAX = 1000 )
*
*     .. Local Scalars ..
      INTEGER          INFO, LWORK
*
*     .. Local Arrays ..
      DOUBLE PRECISION A( LDA, N ), W( N ), WORK( LWMAX )
      DATA             A/
     $ 0.6427421496861863, -0.4658213331496202, 
     $-0.0663840448532566,  0.0907113787628938,
     $-0.4658213331496202,  0.3375996338862058, 
     $ 0.0481112126978231, -0.0657422193453504,
     $-0.0663840448532566,  0.0481112126978231, 
     $ 0.0068563130848518, -0.0093689020386119,
     $ 0.0907113787628938, -0.0657422193453504, 
     $-0.0093689020386119,  0.0128022633043168
c     $ 0.64274215, -0.46582133, -0.06638404,  0.09071138,
c     $-0.46582133,  0.33759963,  0.04811121, -0.06574222,
c     $-0.06638404,  0.04811121,  0.00685631, -0.0093689 ,
c     $ 0.09071138, -0.06574222, -0.0093689 ,  0.01280226
c     $  1.96, 0.00, 0.00, 0.00, 0.00,
c     $ -6.49, 3.80, 0.00, 0.00, 0.00,
c     $ -0.47,-6.39, 4.17, 0.00, 0.00,
c     $ -7.20, 1.50,-1.51, 5.70, 0.00,
c     $ -0.65,-6.34, 2.67, 1.80,-7.10
     $                  /
*
*     .. External Subroutines ..
      EXTERNAL         DSYEV
      EXTERNAL         PRINT_MATRIX
*
*     .. Intrinsic Functions ..
      INTRINSIC        INT, MIN
*
*     .. Executable Statements ..
      WRITE(*,*)'DSYEV Example Program Results'
*
*     Query the optimal workspace.
*
      LWORK = -1
      CALL DSYEV( 'Vectors', 'Upper', N, A, LDA, W, WORK, LWORK, INFO )
      LWORK = MIN( LWMAX, INT( WORK( 1 ) ) )
*
*     Solve eigenproblem.
*
      CALL DSYEV( 'Vectors', 'Upper', N, A, LDA, W, WORK, LWORK, INFO )
*
*     Check for convergence.
*
      IF( INFO.GT.0 ) THEN
         WRITE(*,*)'The algorithm failed to compute eigenvalues.'
         STOP
      END IF
*
*     Print eigenvalues.
*
      CALL PRINT_MATRIX( 'Eigenvalues', 1, N, W, 1 )
*
*     Print eigenvectors.
*
      CALL PRINT_MATRIX( 'Eigenvectors (stored columnwise)', N, N, A, 
     $                   LDA )
      STOP
      END
*
*     End of DSYEV Example.
*






*  =============================================================================
*
*     Auxiliary routine: printing a matrix.
*
      SUBROUTINE PRINT_MATRIX( DESC, M, N, A, LDA )
      CHARACTER*(*)    DESC
      INTEGER          M, N, LDA
      DOUBLE PRECISION A( LDA, * )
*
      INTEGER          I, J
*
*      WRITE(*,*)
      WRITE(*,*) DESC
      DO I = 1, M
         WRITE(*,9998) ( A( I, J ), J = 1, N )
      END DO
*
 9998 FORMAT( 11(:,1X,E20.10) )
      RETURN
      END
