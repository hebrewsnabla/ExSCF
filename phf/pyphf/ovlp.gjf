%nproc=1
%mem=200mb
#p hf/3-21g nosymm iop( 3/33=1, 5/33=4)

title

0 1
H 0.0 0.0 0.0
H 0.0 0.0 2.0


