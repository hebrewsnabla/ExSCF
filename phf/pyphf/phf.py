from pyscf import scf

def convert_to_suhf(mf):
    


class SUHF(scf.UHF):
    def __init__(self):
        return self




def build(mf):
    # S transform
    # get dm
    #
    # grid
    #
    # wigner d
    return mf


def kernel(mf):
    #
    while(true):
        get_fock()
        # diag dm , get dm in NO
        get_OmegaMatrices() # Ng, Pg, Gg
        get_xOmega() # Smt, Hmt
        solv_CI()
        get_S2()
        get_YOmega()
        get_Feff()
        get_E()
        Diag_Feff()
        conv = ConvTest()


def get_fock():
    # get coreH in orthomormal AO
    # X! . H . X