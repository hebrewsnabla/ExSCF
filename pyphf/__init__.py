from pyphf import suscf, sudft

__version__ = '0.4.0'


SUHF = suscf.SUHF
SUDFT  = sudft.SUDFT
