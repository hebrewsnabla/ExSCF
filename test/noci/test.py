#import phf
from pyscf import lib
from pyphf import suscf, guess, noci

lib.num_threads(4)

xyz = '''
N -1.535079   -1.050802   -0.265895
H -1.424942   -1.420987    0.683068
H -1.172743   -1.798291   -0.863000
C -0.674001    0.116995   -0.396238
H -0.667420    0.432032   -1.454565
C -1.220146    1.281062    0.444439
C  0.781650   -0.165583   -0.015767
H -1.219679    1.017906    1.515968
H -0.611700    2.187271    0.312390
H -2.257523    1.488947    0.145218
O  1.176250   -1.168621    0.535350
O  1.608113    0.851853   -0.361143
H  2.499646    0.588022   -0.066072
''' #sys.argv[1]
#fch = '../test_carlos/test_uhf.fchk' #sys.argv[2]
bas = '3-21g'


#mf = util.guess_from_fchk(xyz, bas, fch)
mf = guess.gen(xyz, bas, 0, 0)

mf2 = suscf.SUHF(mf)
#mf2.cut_no = False
#mf2.verbose = 6
#mf2.diis_on = True
mf2.max_cycle = 30
mf2.kernel()

#mf4 = pt2.EMP2(mf2)
#mf4.kernel()
noci.test(mf2)