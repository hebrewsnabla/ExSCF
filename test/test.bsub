#BSUB -J test
#BSUB -n 2
#BSUB -R span[ptile=2]
#BSUB -q small
#BSUB -o test.output

g09 test.gjf
