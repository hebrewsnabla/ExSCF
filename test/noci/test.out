#INFO: **** input file is /share/home/srwang/ExSCF/test/noci/test.py ****
#import phf
from pyscf import lib
from pyphf import suscf, guess, noci

lib.num_threads(4)

xyz = '''
N -1.535079   -1.050802   -0.265895
H -1.424942   -1.420987    0.683068
H -1.172743   -1.798291   -0.863000
C -0.674001    0.116995   -0.396238
H -0.667420    0.432032   -1.454565
C -1.220146    1.281062    0.444439
C  0.781650   -0.165583   -0.015767
H -1.219679    1.017906    1.515968
H -0.611700    2.187271    0.312390
H -2.257523    1.488947    0.145218
O  1.176250   -1.168621    0.535350
O  1.608113    0.851853   -0.361143
H  2.499646    0.588022   -0.066072
''' #sys.argv[1]
#fch = '../test_carlos/test_uhf.fchk' #sys.argv[2]
bas = '3-21g'


#mf = util.guess_from_fchk(xyz, bas, fch)
mf = guess.gen(xyz, bas, 0, 0)

mf2 = suscf.SUHF(mf)
#mf2.cut_no = False
#mf2.verbose = 6
#mf2.diis_on = True
mf2.max_cycle = 30
mf2.kernel()

#mf4 = pt2.EMP2(mf2)
#mf4.kernel()
noci.test(mf2)#INFO: ******************** input file end ********************


System: uname_result(system='Linux', node='xn01', release='3.10.0-1127.el7.x86_64', version='#1 SMP Tue Feb 18 16:39:12 EST 2020', machine='x86_64', processor='x86_64')  Threads 4
Python 3.7.4 (default, Aug 13 2019, 20:35:49) 
[GCC 7.3.0]
numpy 1.17.2  scipy 1.4.1
Date: Tue Jun 29 22:48:47 2021
PySCF version 1.7.6a1
PySCF path  /share/home/srwang/pyscf/pyscf
GIT HEAD      ref: refs/heads/master
GIT master branch  80f71dfe77ac5f9caae26788acf76446987635bd

[CONFIG] conf_file /share/home/srwang/.pyscf_conf.py
[INPUT] verbose = 4
[INPUT] num. atoms = 13
[INPUT] num. electrons = 48
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry False subgroup None
[INPUT] Mole.unit = angstrom
[INPUT]  1 N     -1.535079000000  -1.050802000000  -0.265895000000 AA   -2.900878889571  -1.985727991145  -0.502468727891 Bohr
[INPUT]  2 H     -1.424942000000  -1.420987000000   0.683068000000 AA   -2.692750123390  -2.685276256567   1.290811444454 Bohr
[INPUT]  3 H     -1.172743000000  -1.798291000000  -0.863000000000 AA   -2.216163084501  -3.398277482270  -1.630833645500 Bohr
[INPUT]  4 C     -0.674001000000   0.116995000000  -0.396238000000 AA   -1.273677297683   0.221088507943  -0.748781300145 Bohr
[INPUT]  5 H     -0.667420000000   0.432032000000  -1.454565000000 AA   -1.261241010057   0.816422157048  -2.748729480378 Bohr
[INPUT]  6 C     -1.220146000000   1.281062000000   0.444439000000 AA   -2.305741771984   2.420856328588   0.839867989076 Bohr
[INPUT]  7 C      0.781650000000  -0.165583000000  -0.015767000000 AA    1.477104425266  -0.312906520884  -0.029795311806 Bohr
[INPUT]  8 H     -1.219679000000   1.017906000000   1.515968000000 AA   -2.304859269883   1.923563560552   2.864764333605 Bohr
[INPUT]  9 H     -0.611700000000   2.187271000000   0.312390000000 AA   -1.155945470396   4.133343150204   0.590331544053 Bohr
[INPUT] 10 H     -2.257523000000   1.488947000000   0.145218000000 AA   -4.266100189906   2.813702043993   0.274422248357 Bohr
[INPUT] 11 O      1.176250000000  -1.168621000000   0.535350000000 AA    2.222790354020  -2.208373633415   1.011664880786 Bohr
[INPUT] 12 O      1.608113000000   0.851853000000  -0.361143000000 AA    3.038893147353   1.609768868389  -0.682461361804 Bohr
[INPUT] 13 H      2.499646000000   0.588022000000  -0.066072000000 AA    4.723646348365   1.111200535219  -0.124857984502 Bohr

nuclear repulsion = 248.457150642938
number of shells = 44
number of NR pGTOs = 111
number of NR cGTOs = 68
basis = 3-21g
ecp = {}
CPU time:         1.03


******** <class 'pyscf.scf.uhf.UHF'> ********
method = UHF
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /share/home/srwang/ExSCF/test/noci/tmpdgveg4ka
max_memory 4000 MB (current use 103 MB)
number electrons alpha = 24  beta = 24
Set gradient conv threshold to 3.16228e-05
init E= -320.535716061003
  alpha nocc = 24  HOMO = -0.212252037168868  LUMO = 0.108441605238368
  beta  nocc = 24  HOMO = -0.220174311429871  LUMO = 0.10220672594102
cycle= 1 E= -319.840129194561  delta_E= 0.696  |g|= 0.685  |ddm|=  2.4
  alpha nocc = 24  HOMO = -0.352423961983976  LUMO = 0.11688133360035
  beta  nocc = 24  HOMO = -0.352966668760975  LUMO = 0.116885147547828
cycle= 2 E= -319.970035862463  delta_E= -0.13  |g|= 0.514  |ddm|= 1.08
  alpha nocc = 24  HOMO = -0.398804438949164  LUMO = 0.158153065308772
  beta  nocc = 24  HOMO = -0.397955554877439  LUMO = 0.157766344396047
cycle= 3 E= -320.064247440532  delta_E= -0.0942  |g|= 0.0559  |ddm|= 0.487
  alpha nocc = 24  HOMO = -0.392830179087653  LUMO = 0.176384103260042
  beta  nocc = 24  HOMO = -0.392344499147167  LUMO = 0.176288841240457
cycle= 4 E= -320.066145632554  delta_E= -0.0019  |g|= 0.0331  |ddm|= 0.0823
  alpha nocc = 24  HOMO = -0.395304073880171  LUMO = 0.176059859302738
  beta  nocc = 24  HOMO = -0.394979821059313  LUMO = 0.17604972954731
cycle= 5 E= -320.066670068208  delta_E= -0.000524  |g|= 0.00643  |ddm|= 0.0397
  alpha nocc = 24  HOMO = -0.396681389504553  LUMO = 0.174594180246899
  beta  nocc = 24  HOMO = -0.396493832954258  LUMO = 0.174636978887136
cycle= 6 E= -320.066742738392  delta_E= -7.27e-05  |g|= 0.0029  |ddm|= 0.0192
  alpha nocc = 24  HOMO = -0.39637272414224  LUMO = 0.175213680990229
  beta  nocc = 24  HOMO = -0.396269692488622  LUMO = 0.17526461872452
cycle= 7 E= -320.066755832615  delta_E= -1.31e-05  |g|= 0.000949  |ddm|= 0.0089
  alpha nocc = 24  HOMO = -0.396382129080218  LUMO = 0.175422974441743
  beta  nocc = 24  HOMO = -0.396328300083697  LUMO = 0.175463264931661
cycle= 8 E= -320.066757169864  delta_E= -1.34e-06  |g|= 0.000279  |ddm|= 0.00283
  alpha nocc = 24  HOMO = -0.39637798668413  LUMO = 0.175313092256878
  beta  nocc = 24  HOMO = -0.396346692999432  LUMO = 0.175343616186386
cycle= 9 E= -320.066757284897  delta_E= -1.15e-07  |g|= 0.000129  |ddm|= 0.000679
  alpha nocc = 24  HOMO = -0.396343280863716  LUMO = 0.175344545862358
  beta  nocc = 24  HOMO = -0.396325486246063  LUMO = 0.175368680302745
cycle= 10 E= -320.066757318147  delta_E= -3.32e-08  |g|= 8.32e-05  |ddm|= 0.00032
  alpha nocc = 24  HOMO = -0.396346458996069  LUMO = 0.1753398966456
  beta  nocc = 24  HOMO = -0.396341377077379  LUMO = 0.175357063091007
cycle= 11 E= -320.066757341895  delta_E= -2.37e-08  |g|= 5.31e-05  |ddm|= 0.000304
  alpha nocc = 24  HOMO = -0.396337562201977  LUMO = 0.175341479953519
  beta  nocc = 24  HOMO = -0.396338519471601  LUMO = 0.175353684502102
cycle= 12 E= -320.066757354309  delta_E= -1.24e-08  |g|= 4.06e-05  |ddm|= 0.000203
  alpha nocc = 24  HOMO = -0.396340885209174  LUMO = 0.175345338235157
  beta  nocc = 24  HOMO = -0.396343793069202  LUMO = 0.175354249840673
cycle= 13 E= -320.066757361456  delta_E= -7.15e-09  |g|= 3.17e-05  |ddm|= 0.000145
  alpha nocc = 24  HOMO = -0.396340486142266  LUMO = 0.175345950441228
  beta  nocc = 24  HOMO = -0.396343172225621  LUMO = 0.175351430278657
cycle= 14 E= -320.066757368  delta_E= -6.54e-09  |g|= 2.24e-05  |ddm|= 0.00017
  alpha nocc = 24  HOMO = -0.396340122935929  LUMO = 0.17534687684443
  beta  nocc = 24  HOMO = -0.396340845935791  LUMO = 0.17534825846441
cycle= 15 E= -320.066757373046  delta_E= -5.05e-09  |g|= 1.04e-05  |ddm|= 0.000241
  alpha nocc = 24  HOMO = -0.396341682339604  LUMO = 0.175349108020185
  beta  nocc = 24  HOMO = -0.396341625409316  LUMO = 0.175349150153048
cycle= 16 E= -320.066757373651  delta_E= -6.05e-10  |g|= 3.93e-06  |ddm|= 9.31e-05
  alpha nocc = 24  HOMO = -0.396341351138765  LUMO = 0.175348705504812
  beta  nocc = 24  HOMO = -0.396341339754637  LUMO = 0.17534883798039
Extra cycle  E= -320.066757373673  delta_E= -2.25e-11  |g|= 2.51e-06  |ddm|= 7.18e-06
converged SCF energy = -320.066757373673  <S^2> = 5.3656279e-10  2S+1 = 1

******** <class 'pyphf.suscf.SUHF'> ********
Date: Tue Jun 29 22:48:47 2021
pyphf version 0.4.0
pyphf path  /share/home/srwang/ExSCF
GIT ORIG_HEAD 198e48480b188f02e90f0e6e8cbbe5568091e1c6
GIT HEAD (branch main) 920a77f5e310fde7a62df6d54d83e28771900f5b
max_cycle: 30
verbose: 4                # normal
conv_tol: 1e-07           # 1e-07 for RMSD(dm), 1e-05 for MaxD(dm), 1e-09 for dE
chkfile0: 6906_ges.pchk # the file store hf for guess
chkfile:  6906_su.pchk  # the file store suhf info
DIIS: <class 'pyscf.scf.diis.CDIIS'>
diis_start_cyc = 10
grids:  [0.06237655 0.31939555 0.74529195 1.28265786 1.85893479 2.39630071
 2.8221971  3.07921611] 
weights:  [0.15900941 0.34931531 0.49276925 0.56970235 0.56970235 0.49276925
 0.34931531 0.15900941]
S = 0.0, Sz = 0.0
Wigner small d:  1
value : [1. 1. 1. 1. 1. 1. 1. 1.]
time for Preparation before cyc: 0.105
**** Start Cycle 1 ****
time for NO, Ng: 0.141
time for Gg: 0.771
detC -0.9999999999999875
detNg [1. 1. 1. 1. 1. 1. 1. 1.]
xg [1. 1. 1. 1. 1. 1. 1. 1.]
ciS 1.9999999994633777
ciH -1137.0478157282712
S2 = 0.000000
Hsp + Hph =  -568.5239080166768
Alpha occ 24 vir 44; Beta occ 24 vir 44
Alpha energies:  [-0.542127 -0.523173 -0.502179 -0.472511 -0.45321  -0.396342] <- HOMO
         LUMO->  [0.175349 0.253896 0.265425 0.299545 0.320094 0.343042]
Beta energies:   [-0.542127 -0.523173 -0.50218  -0.472507 -0.453209 -0.396342] <- HOMO
         LUMO->  [0.175349 0.253896 0.265425 0.299545 0.320094 0.343042]
time for xg, H, S2, Yg, Feff: 0.043
 E =   -320.06675737
**** Start Cycle 2 ****
time for NO, Ng: 0.139
time for Gg: 0.759
detC -0.9999999999999925
detNg [1. 1. 1. 1. 1. 1. 1. 1.]
xg [1. 1. 1. 1. 1. 1. 1. 1.]
ciS 1.9999999992159005
ciH -1137.0478155876217
S2 = 0.000000
Hsp + Hph =  -568.5239080167005
Alpha occ 24 vir 44; Beta occ 24 vir 44
Alpha energies:  [-0.542127 -0.523173 -0.502179 -0.472512 -0.453211 -0.396341] <- HOMO
         LUMO->  [0.175349 0.253896 0.265425 0.299545 0.320094 0.343042]
Beta energies:   [-0.542127 -0.523173 -0.50218  -0.472506 -0.453209 -0.396341] <- HOMO
         LUMO->  [0.175349 0.253896 0.265425 0.299545 0.320094 0.343042]
time for xg, H, S2, Yg, Feff: 0.040

***************
SUHF converged at cycle 2
Final E(SUHF) =   -320.06675737, delta E = -2.36469e-11, MaxD(dm) = 2.21684e-06, RMSD(dm) = 6.30943e-08
time for cyc: 1.893
******* suhf density *****
Clebsch-Gordan coeff: < 0 0 0 0 | 0 0 > = 1
Clebsch-Gordan coeff: < 0 0 1 0 | 0 0 > = 0
xggint 1.0
time for 1pdm: 0.040
SUHF NO occ alpha:  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  ...
SUHF NO occ beta:   1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000  ...
SUHF NO occ total:  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  2.000000  ...
core 24, active 0, external 44
time for natorb: 0.002
***** End of SUHF *****
time tot: 2.040
Date: Tue Jun 29 22:48:49 2021
Excitation: Alpha  [23]  ->  [24]
             Beta  []  ->  []
Former occ pattern: Alpha [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
                     Beta [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
New occ pattern:    Alpha [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
                     Beta [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
xg [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
ciS 0.0
xg [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
ciS 0.0
